test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ClientTests/Tests/ClientQueryTests.cs

```diff
@@ -0,0 +1,228 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ClientQueryTests.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.OData;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.Client.E2E.TestCommon;
+using Microsoft.OData.Client.E2E.Tests.ClientTests.Server;
+using Microsoft.OData.Client.E2E.Tests.Common.Clients.EndToEnd;
+using Microsoft.OData.Client.E2E.Tests.Common.Clients.EndToEnd.Default;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+using Xunit;
+
+namespace Microsoft.OData.Client.E2E.Tests.ClientTests.Tests
+{
+    public class ClientQueryTests : EndToEndTestBase<ClientQueryTests.TestsStartup>
+    {
+        private readonly Uri _baseUri;
+        private readonly Container _context;
+
+        public class TestsStartup : TestStartupBase
+        {
+            public override void ConfigureServices(IServiceCollection services)
+            {
+                services.ConfigureControllers(typeof(ClientQueryTestsController), typeof(MetadataController));
+
+                services.AddControllers().AddOData(opt => opt.EnableQueryFeatures()
+                    .AddRouteComponents("odata", CommonEndToEndEdmModel.GetEdmModel()));
+            }
+        }
+
+        public ClientQueryTests(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+            _baseUri = new Uri(Client.BaseAddress, "odata/");
+            _context = new Container(_baseUri);
+            _context.HttpClientFactory = HttpClientFactory;
+        }
+
+        [Fact]
+        public void DollarFilter_UsingContains_ExecutesSuccessfully()
+        {
+            var testlists = new List<string>()

```

Comment:
```suggestion
            var testLists = new List<string>()

```
Consider using `[Theory]` instead of the for loop.